lib/merchant_repository.rb

```diff
@@ -39,6 +39,36 @@ def average_items_per_merchant_standard_deviation
     deviation(number_of_items_per_merchant, average_items_per_merchant).round(2)
   end
 
+  def number_of_invoices_per_merchant
+    all.map do |merchant|
+      merchant.invoice_count
+    end
+  end
+
+  def average_invoices_per_merchant
+    average(number_of_invoices_per_merchant.sum, all.length).round(2)
+  end
```

Comment:
I have been migrating all of my 'rounds' to SA; having the repos collect raw data and the SA make the final decisions about what that data should look like is a story that makes more sense to me. This might become more important if we need to manipulate data from multiple Repos. Just a thought, I don't think we have a consistent standard yet in our code base and it might make no difference in the end.

Reply:
That's a really good point! I support that change!